kernel/aster-nix/src/time/timer_callback.rs

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::{sync::atomic::AtomicBool, time::Duration};
+
+use atomic::Ordering;
+
+use crate::prelude::*;
+
+/// A `TimerCallback` can be used to execute a timer callback function.
+pub struct TimerCallback {
+    expire_time: Duration,
+    callback: Box<dyn Fn() + Send + Sync>,
+    is_cancelled: AtomicBool,
+}
+
+impl TimerCallback {
+    /// Create an instance of `TimerCallback`.
+    pub fn new(timeout: Duration, callback: Box<dyn Fn() + Send + Sync>) -> Self {
+        Self {
+            expire_time: timeout,
+            callback,
+            is_cancelled: AtomicBool::new(false),
+        }
+    }
+
+    /// Return the expire time of the `TimerCallback`.
+    pub fn expire_time(&self) -> Duration {
+        self.expire_time
+    }
+
+    /// Whether the set timeout is reached
+    pub fn is_expired(&self, now: Duration) -> bool {
+        self.expire_time <= now
+    }
+
+    /// Cancel a `TimerCallback`. If the callback function has not been called,
+    /// it will never be called again.
+    pub fn cancel(&self) {
+        self.is_cancelled.store(true, Ordering::Release);
+    }
+
+    // Whether the `TimerCallback` is cancelled.
+    pub(super) fn is_cancelled(&self) -> bool {
```

Comment:
Method description should start with a verb. 
So the doc should be `Return Whether the `TimerCallback` is cancelled`. Please should also examine other docs in this PR.